### DISCUSSION TITLE
请问插件的生命周期是怎样的？如何实现持久的 Websocket 链接

### DISCUSSION THREAD

**Discussion by XKaguya** [nonebot/discussions#544] [问答] [answered]
如题，主要想问的是如何实现插件加载后建立一个持久的 Websocket 等链接

**Comment by yanyongyu** [comment]
没懂你的需求是什么，bot 的 ws 连接不是由插件建立的。

**Reply by XKaguya** [reply]
> 没懂你的需求是什么，bot 的 ws 连接不是由插件建立的。

不是 bot 的 ws 连接，我的需求是插件能建立一个持久性的 Websocket 连接，就是不需要让插件每次都重新连接到对应 ws 来发送/读取

**Reply by yanyongyu** [reply]
还是没太懂，如果你要建立一个持久性的连接，那么在 driver startup 的时候进行连接，shutdown 的时候断开连接就行了？什么叫每次都进行连接

**Reply by XKaguya** [reply]
> 还是没太懂，如果你要建立一个持久性的连接，那么在 driver startup 的时候进行连接，shutdown 的时候断开连接就行了？什么叫每次都进行连接

嗯可能我描述有点问题，我本身对这块不是很了解，想尝试从文档找到类似实现方式没找到..
就是 目前我所有插件的实现方法都是每次需要用到时 nb 端发起一次连接到 ws 服务器，我想要就是建立一个从 bot 载入插件到某一刻出现问题都可用的一个 websocket 连接

**Reply by yanyongyu** [reply, marked as answer]
那你需要的就是我上面说的，在driver startup hook的时候建立一个ws连接

**Reply by XKaguya** [reply]
> 那你需要的就是我上面说的，在driver startup hook的时候建立一个ws连接

好的，谢谢大佬
